library/src/main/java/org/apache/apex/malhar/lib/state/spillable/SpillableStateStore.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.apex.malhar.lib.state.spillable;
+
+import org.apache.apex.malhar.lib.state.BucketedState;
+
+import com.datatorrent.api.Component;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.Operator;
+
+public interface SpillableStateStore extends BucketedState, Component<Context.OperatorContext>,
```

Comment:
The WindowedOperator only needs a storage that it can grab data from, and the interface of which is defined in the WindowedStorage interfaces. The user of WindowedOperator can set the storage implementation before setup().  That's why it does not have any knowledge of Spillable components. Take a look at the interface WindowedStorage here: https://github.com/davidyan74/apex-malhar/blob/66da57e08e31edb15e83a050f3940dd276851c23/library/src/main/java/org/apache/apex/malhar/lib/window/WindowedStorage.java

If user of the WindowedOperator wants a custom storage implementation, they can simply implement those interfaces without having to deal with Spillable components.